index.js

```diff
@@ -98,6 +102,8 @@ module.exports = opts => {
 			});
 		}
 	};
-
+	if (hasYarn(pkgCwd)) {
```

Comment:
And if the `hasYarn` is made async, it would look like this.

```js
if (opts.skipInstall) {
    return Promise.resolve(post);
}

return hasYarn(pkgCwd).then(hasYarn => {
    if (hasYarn) {
        return execa('yarn', ['add', '--dev', 'xo'], {cwd: pkgCwd}).then(post);
    }

    return execa('npm', ['install', '--save-dev', 'xo'], {cwd: pkgCwd}).then(post);
});
```